base/cvd/cuttlefish/host/commands/cvd/cli/frontline_parser.h

```diff
@@ -39,46 +32,7 @@ Result<cvd_common::Args> ExtractCvdArgs(cvd_common::Args& args);
  *     b) non-selector flags
  *  3. subcommand
  *  4. subcommand arguments
- *
- * This is currently on the client side but will be moved to the server
- * side.
  */
-class FrontlineParser {
-  using ArgumentsSeparator = selector::ArgumentsSeparator;
-
- public:
-  struct ParserParam {
-    // commands supported by the server
-    std::vector<std::string> server_supported_subcmds;
-    cvd_common::Args all_args;
-  };
-
-  // This call must guarantee all public methods will be valid
-  static Result<std::unique_ptr<FrontlineParser>> Parse(ParserParam param);
-
-  const std::string& ProgPath() const;
-  std::optional<std::string> SubCmd() const;
-  const cvd_common::Args& SubCmdArgs() const;
-  const cvd_common::Args& CvdArgs() const;
-
- private:
-  FrontlineParser(const ParserParam& parser_param);
-
-  // internal workers in order
-  Result<void> Separate();
-  Result<cvd_common::Args> ValidSubcmdsList();
-  Result<std::unique_ptr<ArgumentsSeparator>> CallSeparator();
-  struct FilterOutput {
-    bool clean;
-    bool help;
-    cvd_common::Args selector_args;
-  };
-  Result<FilterOutput> FilterNonSelectorArgs();
-
-  cvd_common::Args server_supported_subcmds_;
-  const cvd_common::Args all_args_;
-  const std::vector<std::string> internal_cmds_;
-  std::unique_ptr<ArgumentsSeparator> arguments_separator_;
-};
+Result<cvd_common::Args> ExtractCvdArgs(cvd_common::Args& args);
```

Comment:
Please update the comment over this to explicitly mention that it returns the cvd arguments after removing them from the input parameter.
That comment seems to have been originally for the FrontlineParser class, which did more than this function.